### PR TITLE
Ewl-870: sticky player component

### PR DESCRIPTION
**Pull request by val3ntin**
this enables sticky player .

* it can be sticked to `top` / `bottom`
* it sticks only when player is in `playing` mode
* it does not have animation when it's being un-sticked as it might take some time to implement ( track scroll position + direction for proper animation )

stick to top:
Screen-Recording-2019-07-10-at-9 11 54-AM

stick to bottom:
Screen-Recording-2019-07-10-at-9 12 10-AM

### CODE REVIEW

File: src/components/JwPlayer/index.js [hunk 1]
```diff
@@ -1,10 +1,180 @@
...
+  styledWrapperRef = React.createRef();
+
+  state = {
+    isPlaying: false,
+    isPlayerSticky: false,
+    isStickyPanelClosed: false,
+    minWrapperHeight: null,
+  };
+
+  handleStickyOnScroll = debounce(() => {
```
Comment (val3ntin) [inline src/components/JwPlayer/index.js]: one of improvements here can be using `rAF` logic for window.requestAnimationFrame once trying to stick the panel. that would probably make the transition even better.
can be considered as a future improvement.

File: src/components/JwPlayer/index.js [hunk 2]
```diff
@@ -1,10 +1,180 @@
...
+      flex-flow: nowrap row;
+      align-items: center;
+      width: 100%;
+      max-width: 355px;
+      height: 67px;
+      color: ${brandPlus2};
+      border-radius: 4px;
+      overflow: hidden;
+      transform: translate3d(0, 0, 0);
+      box-shadow: 0 0 30px 10px rgba(0,0,0,0.2);
```
Comment (val3ntin) [inline src/components/JwPlayer/index.js]: btw we don't have black color in order to use our color set with `polished`'s `rgba`

Comment (val3ntin) [reply src/components/JwPlayer/index.js]: thus it's usual rgba palette with hardcoded black color.

File: src/components/JwPlayer/index.js [hunk 3]
```diff
@@ -1,10 +1,180 @@
...
+      
+      > div[id = ${playerId}] {
```
[the remark this answers is not in the document]
Comment (val3ntin) [reply src/components/JwPlayer/index.js]: :) indeed. done

File: src/components/JwPlayer/index.js [hunk 4]
```diff
@@ -1,10 +1,180 @@
...
+      animation: slide-from-${props.stickTo} 250ms ease-out forwards;
+      flex-flow: nowrap row;
+      align-items: center;
+      width: 100%;
+      max-width: 355px;
+      height: 67px;
+      color: ${brandPlus2};
+      border-radius: 4px;
+      overflow: hidden;
+      transform: translate3d(0, 0, 0);
```
[the remark this answers is not in the document]
Comment (val3ntin) [reply src/components/JwPlayer/index.js]: yes, but not in this place but on the root of `StyledPlayer` so browser knows it once the div is renderred.

File: src/components/JwPlayer/index.js [hunk 5]
```diff
@@ -1,10 +1,180 @@
...
+        max-width: 534px;
+      `)}
+      
+      ${wide(css`
+        height: 104px;
+        max-width: 552px;
+      `)}
+    `}
+
+  @keyframes slide-from-bottom {
```
[the remark this answers is not in the document]
Comment (val3ntin) [reply src/components/JwPlayer/index.js]: thx for your tip, however it won't be so flexible in the way I used it to provide the animation name:
```css
animation: slide-from-${props.stickTo}
```

by using helper I would then need to modify the code in a bit mess manner :) 

smth like 
```
animaition-name: ${props => props.stickTo === 'bottom' ? keyframeBottomAnimation : otherAnimation ...}

```
didn't want that to happen :)

File: src/components/JwPlayer/index.js [hunk 6]
```diff
@@ -1,10 +1,180 @@
...
+  top: 50%;
+  right: 20px;
+  transform: translateY(-50%);
+  width: 17px;
+  height: 17px;
+  background: url(${Cross}) no-repeat center center;
+  cursor: pointer;
+`;
+
+export const StyledStickyCotent = styled.div`
```
[the remark this answers is not in the document]
Comment (val3ntin) [reply src/components/JwPlayer/index.js]: thx

File: src/components/JwPlayer/index.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (val3ntin) [reply src/components/JwPlayer/index.js]: done ;)

File: src/components/JwPlayer/index.js [hunk 7]
```diff
@@ -1,10 +1,181 @@
...
+import debounce from 'lodash/debounce';
+import Cross from 'src/assets/close-cross.svg';
 import ScriptInjector from '../ScriptInjector';
+import { medium, large, wide } from '../../breakpoints';
+import { brandPlus2, coreLightMinus1 } from '../../colors';
 
-const playerId = 'eurosport-web-player';
+export const playerId = 'eurosport-web-player';
+
+export const ONE_FRAME = 16.7;
```
Comment (val3ntin) [inline src/components/JwPlayer/index.js]: 1000ms / 60(fps) ^^

File: src/components/JwPlayer/index.js (hunk 2 above)
Comment (leozero) [reply src/components/JwPlayer/index.js]: You're right, I'm also for adding white and black and white in our colors

File: src/components/JwPlayer/index.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (arturkin) [reply src/components/JwPlayer/index.js]: Not sure `emotion` is better then sass, but we are stilll using it 😜

**Pull request merged**